pkg/brownfield/ports.go

```diff
@@ -39,10 +40,13 @@ func (er ExistingResources) GetBlacklistedPorts() ([]n.ApplicationGatewayFronten
 
 // MergePorts merges list of lists of ports into a single list, maintaining uniqueness.
 func MergePorts(portBuckets ...[]n.ApplicationGatewayFrontendPort) []n.ApplicationGatewayFrontendPort {
-	uniq := make(portsByName)
+	uniq := make(map[int32]n.ApplicationGatewayFrontendPort)
```

Comment:
This is good. You could probably re-design the exsting type (or not).
This could also go in asap.